### PR TITLE
An event store as an example of Storage + Sherlock = <3

### PR DESCRIPTION
**Pull request by dkorolev**

### CODE REVIEW

File: examples/EventStore/event_store.h [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+// NOTE: This class assumes the `events` storage field has the type `EVENT_TYPE`.
+// NOTE: This class assumes `EXTRA_TYPE` can be constructed from a `const EVENT_TYPE&`.
+template <template <template <typename...> class, template <typename> class, typename>
+          class CURRENT_STORAGE_TYPE,
+          typename EVENT_TYPE,
+          typename EXTRA_TYPE,
+          template <typename...> class DB_PERSISTER>
+struct EventStore final {
+  // TODO(dkorolev) + TODO(mzhurovich): All three template parameters here are only to extract `T_TRANSACTION`.
+  // Factor it out?
```
Comment (mzhurovich) [inline examples/EventStore/event_store.h]: Good idea.

File: examples/EventStore/event_store.h [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+// TODO(dkorolev): Split into test and "live" demo.
+
+#include "../../TypeSystem/struct.h"
+#include "../../Storage/storage.h"
+#include "../../Storage/persister/sherlock.h"
+
+#include "../../Blocks/HTTP/api.h"
+
+// TODO(dkorolev) + TODO(mzhurovich): Convert all `T_UPPER_CASE` into `upper_case_t`?
```
Comment (mzhurovich) [inline examples/EventStore/event_store.h]: Everywhere in Current? Let's find time for this in the next couple of weeks.

File: examples/EventStore/test.cc [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+#include "event_store.h"
+#include "schema.h"
+
+#include "../../Bricks/dflags/dflags.h"
+#include "../../Bricks/file/file.h"
+
+#include "../../3rdparty/gtest/gtest-main-with-dflags.h"
+
+#ifndef _MSC_VER
```
Comment (mzhurovich) [inline examples/EventStore/test.cc]: Don't we want to switch to `CURRENT_WINDOWS` instead of `MSC_VER`?

**Comment by mzhurovich** [comment]
@dkorolev LGTM! Let's sync up, when you're online.

**Comment by dkorolev** [comment]
@mzhurovich PTaL!

**Pull request merged**